ratpack-pac4j/src/main/java/ratpack/pac4j/internal/Pac4jAuthenticator.java

```diff
@@ -81,13 +82,28 @@ public void handle(Context ctx) throws Exception {
   }
 
   public Clients createClients(Context ctx, PathBinding pathBinding) throws Exception {
+    String callback = createCallbackUrl(ctx.get(PublicAddress.class), pathBinding, path);
+
+    return new Clients(callback, getClientsFromProvider(ctx, clientsProvider));
+  }
+
+  private String createCallbackUrl(PublicAddress publicAddress, PathBinding pathBinding, String path) {
     String boundTo = pathBinding.getBoundTo();
-    PublicAddress publicAddress = ctx.get(PublicAddress.class);
-    String absoluteCallbackUrl = publicAddress.get() + boundTo + "/" + path;
+    URI address = publicAddress.get();
+
+    if(!boundTo.isEmpty()) {
+      if (!(address.toString().endsWith("/") || boundTo.startsWith("/"))) {
+        boundTo = "/" + boundTo;
+      }
+    }
+
+    return address + boundTo + "/" + path;
+  }
 
+  @SuppressWarnings("rawtypes")
```

Comment:
Is there a way to avoid having this warning at all? What about using a wildcard generic type for `Client` here? Is there an issue with that I'm not seeing?